components/board/genericlinux/board.go

```diff
@@ -588,7 +588,14 @@ func (b *sysfsBoard) GPIOPinByName(pinName string) (board.GPIOPin, error) {
 	// Otherwise, the pins are stored in b.gpios.
 	pin, ok := b.gpios[pinName]
 	if !ok {
```

Comment:
I'd flip these if statements around. This lets you return early when you have the answer, and will keep the code less indented (which makes it easier to read/understand/maintain).

```go
if pin, ok := b.gpios[pinName]; ok {
    return pin, nil
}
// Check if the pin is a digital interrupt
if interrupt, ok := b.interrupts[pinName]; ok {
    return &gpioInterruptWrapperPin{*interrupt}, nil
}
return nil, errors.Errorf("cannot find unknown pin %s", pinName)
```